sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -278,17 +278,13 @@ function getDecodedResponseStream(
 
 function streamToText(stream: NodeJS.ReadableStream): Promise<string> {
   return new Promise<string>((resolve, reject) => {
-    const buffer: string[] = [];
+    const buffer: Buffer[] = [];
 
     stream.on("data", (chunk) => {
-      if (typeof chunk === "string") {
-        buffer.push(chunk);
-      } else {
-        buffer.push(chunk.toString());
-      }
+      buffer.push(Buffer.from(chunk));
```

Comment:
Aren't chunks already Buffers unless you set the encoding?

Reply:
https://nodejs.org/docs/latest-v8.x/api/stream.html#stream_event_data Could be strings too

Reply:
> The listener callback will be passed the chunk of data as a string if a default encoding has been specified for the stream using the readable.setEncoding() method; otherwise the data will be passed as a Buffer.

I don't believe we're setting the encoding. My worry is that we're creating twice as many Buffers since Buffer.from(Buffer) copies data.

Reply:
Maybe an alternative is to only push Buffer.from(chunk) if the type of chunk is a string, otherwise just push the chunk as is.

Reply:
That's fair. I updated this to check with `Buffer.isBuffer`